src/NLog/Targets/FileTarget.cs

```diff
@@ -1063,7 +1126,11 @@ private void DateAndSequentialArchive(string fileName, string pattern, LogEventI
                     if (fileIndex > archiveFileCount - this.MaxArchiveFiles)
                         break;
 
-                    File.Delete(filesByDate[fileIndex]);
+                    var file = filesByDate[fileIndex];
+                    File.Delete(file);
+                    var preventNullRefs = this.OnArchiveFileDeleted;
+                    if (preventNullRefs != null)
+                        preventNullRefs(file);
```

Comment:
Code duplication. Should be call to `fileArchive_OnArchiveFileDeleted` / `fileArchive_OnFileArchived`?